### ISSUE TITLE
Get value from datepicker

### ISSUE THREAD

**Issue by kenanchristian** [react-materialize/react-materialize#258]
Hi there, I'm successfully created a datepicker using the solution from issue #177, which I modified become like this
```js
class App extends React.Component {
   constructor(){
     super();
    this.handleDateChange = this.handleDateChange.bind(this);
   }
    handleDateChange(e){
        this.setState({roomData: { ...this.state.roomData, date: e.currentTarget.value }})
    }
  componentDidMount () {
    $('.datepicker').pickadate({
      selectMonths: true, // Creates a dropdown to control month
      selectYears: 15 // Creates a dropdown of 15 years to control year
    });
  }

  render () {
        let today = new Date().toISOString().split("T")[0];
        let _this = this;
    return (
      <Input type='date' className='datepicker' data-value={today} onChange={this.handleDateChange}/>
    )
  }
};
```

Now how can I get the value of the set date from it? The onChange handler doesn't seem to work

**Comment by yurikilian** [comment]
Hi. This doesn't worked for me. What I do was create a new component using input field.


See this code below: 

`import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import idgen from 'react-materialize/lib/idgen';

export default class DatePicker extends React.Component {
    constructor(props) {
        super(props);
        this.renderFormat = 'dd !de mmmm !de yyyy';

        let defaultValue = this.props.defaultValue || '';
        this.state = { value: defaultValue,  renderedValue: defaultValue};
       
    }
    componentDidMount() {
        let diaSemana = [ 'Domingo', 'Segunda-Feira', 'Terca-Feira', 'Quarta-Feira', 'Quinta-Feira', 'Sexta-Feira', 'Sabado' ];
        let mesAno = [ 'Janeiro', 'Fevereiro', 'Marco', 'Abril', 'Maio', 'Junho', 'Julho', 'Agosto', 'Setembro', 'Outubro', 'Novembro' , 'Dezembro' ];
        const component = this;

        window.$(this.refs.datepicker).pickadate({
            monthsFull: mesAno,
            weekdaysFull: diaSemana,
            monthsShort: [ 'Jan', 'Fev', 'Mar', 'Abr', 'Mai', 'Jun', 'Jul', 'Ago', 'Set', 'Out', 'Nov', 'Dez' ],
            weekdaysShort: [ 'Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sab' ],
            weekdaysLetter: [ 'D', 'S', 'T', 'Q', 'Q', 'S', 'S' ],
            selectMonths: true,
            selectYears: true,
            hiddenName: true,
            format: this.renderFormat,
            today: "Hoje",
            close: "Fechar",
            clear: "Limpar",
            onStart:function(){
                if(component.state.value && component.state.value.isEmpty.length !== 0) {
                    this.set( 'select', component.state.value);
                }
            },
            onSet: function(e) {   
                component.onChange({target: {value: new Date(e.select), renderedValue: this.get(this.renderFormat)}});
                this.close();
            }, 
            onClose(e) {
                window.$(document.activeElement).blur();
            }
        });
    }

    renderIcon() {
        const {children} = this.props;
        let icon = null;
        if (React.Children.count(children) === 1) {
            icon = React
                .Children
                .only(children);
        }
        return icon === null
            ? null
            : React.cloneElement(icon, {className: 'prefix'});
    }

    renderLabelIfExists(id) {
        if (this.props.label) {
            return <label htmlFor={id}>{this.props.label}</label>
        }
    }

    appendSizeClassNames(classNames) {
         if(this.props.s) {
            classNames.push(['s' + this.props.s]);
        } 

        if(this.props.l) {
            classNames.push(['l' + this.props.l]);
        } 

        if(this.props.m) {
            classNames.push(['m' + this.props.m]);
        } 
    }

    triggerComponentOnChange(event) {
        delete event.target.renderedValue;
        event.target.name = this.props.name;
        this.props.onChange(event);
    }

    onChange(event) {
         this.setState({value: event.target.value, renderedValue: event.target.renderedValue});
         this.triggerComponentOnChange(event);    
    }

    render() {       
        let classNames = ['col', 'input-field'];
        this.appendSizeClassNames(classNames);

        const _id = 'datepicker-' + idgen();

        return (
            <div className={cx(classNames)}>
                {this.renderIcon()}
                <input type="text" id={_id} ref="datepicker" value={this.state.renderedValue} onChange={(event) => this.onChange(event)}/> 
                {this.renderLabelIfExists(_id)}
            </div>
        )
    }
}

DatePicker.propTypes = {
    s: PropTypes.number,
    m: PropTypes.number,
    l: PropTypes.number,
    className: PropTypes.string,
    label: PropTypes.node,
    error: PropTypes.string,
    success: PropTypes.string,
    defaultValue: PropTypes.instanceOf(Date),
    placeholder: PropTypes.string,
    id: PropTypes.string,
    icon: PropTypes.string,
    name: PropTypes.string,
    validate: PropTypes.bool,
    multiple: PropTypes.bool,
    browserDefault: PropTypes.bool,
    onLabel: PropTypes.string,
    offLabel: PropTypes.string,
    onChange: PropTypes.func
};`

Try to use onSelect instead or create a new component.

**Comment by alextrastero** [comment]
Hey @kenanchristian, materialize's datepicker is a modified `pickadate.js` and you can add all the methods pickadate has, like `onSet` which is the callback for when a date has been chosen. Hook your `handleDateChange` to that and it should work :)

```js
export default class App extends Component {
  constructor(){
    super();
    this.handleDateChange = this.handleDateChange.bind(this);
    this.state = {
      roomData: {}
    }
  }
  handleDateChange(e){
    this.setState({roomData: { ...this.state.roomData, date: e.select }})
  }
  componentDidMount () {
    $('.datepicker').pickadate({
      selectMonths: true, // Creates a dropdown to control month
      selectYears: 15, // Creates a dropdown of 15 years to control year
      onSet: this.handleDateChange
    });
  }

  render () {
    let today = new Date().toISOString().split("T")[0];
    let _this = this;
    return (
      <Input type='date' className='datepicker' data-value={today} />
    )
  }
};
```

**Comment by alextrastero** [comment]
I've create an issue to fix this #260

**Comment by alextrastero** [comment]
PR https://github.com/react-materialize/react-materialize/pull/261

**Comment by kenanchristian** [comment]
@alextrastero Thanks for the solution!
Got another strange thing though. My datepicker `data-value` doesn't get updated after on page render,
is this an expected behavior?

**Comment by alextrastero** [comment]
@kenanchristian check out the pickadate API here: http://amsul.ca/pickadate.js/api/ maybe there's some method you can use

**Comment by kenanchristian** [comment]
@alextrastero Yeah I'm aware of the `set select` is available on the API, but I can't seem to get it work.
Is there a way to pass a ref to the component? or maybe there's another workaround to access the DOM when it's already replaced with the datepicker?

**Comment by alextrastero** [comment]
Could you describe the steps to reproduce and what the outcome should be?

**Comment by kenanchristian** [comment]
@alextrastero This is my complete code snippet
https://gist.github.com/kenanchristian/7e85c917df0eaf86efa4aa63e4af094d

I know that passing the component to the state seems like a bad idea, and I still failed to set the date.
I've also tried to update it inside `componentDidUpdate` cycle, but no luck there. Tried passing a `ref` to the datepicker element, it seems that the datpicker replace the original element on render so the `ref` is not included inside the datpicker.

What I want to achieve is, make the datepicker value changes when the component updated, found this in pickadate docs http://amsul.ca/pickadate.js/api/#method-set-select
